azuredevops/internal/service/feed/data_feed.go

```diff
@@ -0,0 +1,79 @@
+package feed
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/feed"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+)
+
+func DataFeed() *schema.Resource {
+	return &schema.Resource{
+		Read: dataFeedRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Optional:     true,
+			},
+			"feed_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				ConflictsWith: []string{
+					"name",
+				},
+			},
+			"project_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Optional:     true,
+			},
+		},
+	}
+}
+
+func dataFeedRead(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	name := d.Get("name").(string)
+	id := d.Get("feed_id").(string)
+	projectId := d.Get("project_id").(string)
+
+	if name == "" && id == "" {
```

Comment:
The constraint can be implemented through `AtLeastOneOf`

Reply:
updated